Sources/NIOHTTP1/NIOHTTPClientUpgradeHandler.swift

```diff
@@ -326,21 +326,21 @@ public final class NIOHTTPClientUpgradeHandler: ChannelDuplexHandler, RemovableC
         // upgrader code is done, we do our final cleanup steps, namely we replay the received data we
         // buffered in the meantime and then remove ourselves from the pipeline.
         let pipeline = context.pipeline
-        let loopBoundContext = context.loopBound
         return {
             self.upgradeState = .upgrading
 
+            // assumeIsolated is safe here because this is only called from channelRead
```

Comment:
Isn't the justification here that `removeHTTPHandlers` returns us a future on the right EL? Should it be modified to return an isolated ELF? Otherwise the guarantee is based on documentation vs types.